services/vpm/vpm-version.service.js

```diff
@@ -0,0 +1,81 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { latest, renderVersionBadge } from '../version.js'
+import { BaseJsonService, NotFound } from '../index.js'
+
+const queryParamSchema = Joi.object({
+  repositoryUrl: optionalUrl.required(),
+  sort: Joi.string().valid('defined', 'semver').default('defined'),
+}).required()
+
+const schema = Joi.object({
+  packages: Joi.object()
+    .pattern(
+      /./,
+      Joi.object({
+        versions: Joi.object().pattern(/./, Joi.object()).required(),
```

Comment:
We should make this

```js
versions: Joi.object().pattern(/./, Joi.object()).min(1).required(),
```

This will require version to have at least one key. At the moment, `Object.keys(pkg.versions)` could be `[]`.